### DISCUSSION TITLE
SSO via OpenAM

### DISCUSSION THREAD

**Discussion by queza85** [exoplatform/exo-community#41] [Q&A]
Hi!
I am trying to install openAM to use SSO.

So I installed as follows version based on Window Server
- exo platform: 6.4 (localhost:8080)
- OpenAM: 11.0.4 (localhost:8888, guide recommend 11.0.0 but they don't provide anymore)
- tomcat7.0.109&java 7u80

As follow guideline for 'non-callback' case, I setup openAM and create exo DB and root user.

I added exo.properties using sample property file in ( exohome in docker: /opt/exo/gatein/conf/exo.properties)

#SSO
gatein.sso.enabled=true
gatein.sso.callback.enabled=${gatein.sso.enabled}
gatein.sso.login.module.enabled=${gatein.sso.enabled}
gatein.sso.login.module.class=org.gatein.sso.agent.login.SSOLoginModule
gatein.sso.server.url=http://localhost:8888/openam
gatein.sso.openam.realm=exo
gatein.sso.portal.url=http://localhost:8080
gatein.sso.filter.logout.class=org.gatein.sso.agent.filter.OpenSSOLogoutFilter
gatein.sso.filter.logout.url=${gatein.sso.server.url}/UI/Logout
gatein.sso.filter.login.sso.url=${gatein.sso.server.url}/UI/Login?realm=${gatein.sso.openam.realm}&goto=${gatein.sso.portal.url}/@@portal.container.name@@/initiatessologin

<img width="640" alt="image" src="https://github.com/exoplatform/exo-community/assets/2861131/756986d6-aa6d-400b-9761-a68db3560262"><br/>:::running openam on localhost:8888
<br/>

![image](https://github.com/exoplatform/exo-community/assets/2861131/9cf5ff0c-34e0-4209-83cc-53e18c041324)
<br/>:::working fine without sso config
<br/>

After restart docker and tomcat, I expect to login via Realm User.
When I access the portal home, it has 404 error even OpenAM is running on port 8888
<img width="640" alt="image" src="https://github.com/exoplatform/exo-community/assets/2861131/355e4cb2-18dd-46b7-b562-959aad704994"><br/>:::404 error with sso config
<br/>

1. So do I need to setup more? 

2. Guideline mentioned "the user must be registered in both the external datasource and eXo Platform". 
So if I added some user in eXoPlatform, the user will be generated automatically in Realm, right?
I though eXoPlatform and Realm has to be synchronised.

**Comment by rdenarie** [comment]
Hello @queza85 
I'm afraid that OpenAM sso is no more supported since long time. 

You should try to use SAML of Openidconnect as Identity Provider.

In addition, I think you find informations you provide in an old documentation. How did you find it ?
The up to date documentation is :
https://exoplatform.github.io

Thanks

**Reply by queza85** [reply]
@rdenarie  Thanks for your information.

I found this document from (https://exo-documentation.readthedocs.io/en/latest/)

So your service didn't provide any more CAS, OpenAM and SPNEGO for SSO?
Where can I refer in your document if I want to setup SAML2?

I would like to handle user information on my own database. Can I get any advise for this?

**Comment by rdenarie** [comment]
Hello

For SAML documentation, you can follow this : https://exo-documentation.readthedocs.io/en/latest/SSO.html?highlight=saml#saml2
For Oauth SSO or openid, you can follow this : https://docs.exoplatform.org/administration/oauth-integration.html#overview

We have not migrate the saml documentation in the new documentation.

So, I confirm that CAS, OpenAM and SPNEGO are no more supported for SSO.

If case of SAML, you can plug eXo on your LDAP/AD in order to have your users available in exo (https://docs.exoplatform.org/administration/LDAP.html#introduction)
In case of Oauth/OpenId, you can also plug LDAP/AD, but it is not mandatory as there is a "On The fly" registration process for these SSO (if user is not known, he is automatically registred on successful authentication)

**Reply by queza85** [reply]
thank you for your inform. 
I will try to use KeyCloak based on SAML.
